scripts/tests/run_tv_casting_test.py

```diff
@@ -0,0 +1,225 @@
+                if "Discovered Commissioner" in next_line:
+                    break
+
+            # Append the extracted information to the devices list
+            discovered_commissioners.append({
+                "discovered_commissioner": commissioner,
+                "vendor_id": vendor_id,
+                "product_id": product_id,
+                "device_type": device_type
+            })
+
+    # If the list of discovered commissioners is empty and we didn't find the "No commissioner discovered" string,
+    # then something went wrong. Exit on error.
+    if len(discovered_commissioners) == 0:
+        print('No commissioner(s) discovered! The list of discovered commissioner(s) is empty!')
+        handle_discovery_failure()
+
+
+# Test if the Linux tv-casting-app is able to discover the Linux tv-app.
+# The Linux tv-casting-app and the tv-app will be run in separate processes.
+# Their corresponding output will be written to their respective log files.
+# The log file of the tv-casting-app will be parsed for strings of interest
+# which will be printed to the console.
+def test_discovery_fn():
+
+    with open(LINUX_TV_APP_LOGS, 'w') as fd1, open(LINUX_TV_CASTING_APP_LOGS, 'w') as fd2:
+
+        # Run the Linux tv-app and write the output to file
+        tv_app_rel_path = 'out/tv-app/chip-tv-app'
+        tv_app_abs_path = os.path.abspath(tv_app_rel_path)
+        subprocess.Popen(tv_app_abs_path, stdout=fd1, stderr=subprocess.PIPE, text=True)
+
+        time.sleep(RUN_INTERVAL)
+
+        # Run the Linux tv-casting-app and write the output to file
+        tv_casting_app_rel_path = 'out/tv-casting-app/chip-tv-casting-app'
+        tv_casting_app_abs_path = os.path.abspath(tv_casting_app_rel_path)
+        tv_casting_app_process = subprocess.Popen(
+            tv_casting_app_abs_path, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+
+        for line in tv_casting_app_process.stdout:
+            # Write the line to the Linux tv-casting-app log file
+            fd2.write(line)
+
+            # Fail fast if "No commissioner discovered" string found
+            if "No commissioner discovered" in line:
+                print(line)
+                handle_discovery_failure()
+
+            # Check if the Vendor ID, Product ID, and Device Type match the expected values
+            if "Vendor ID:" in line:
+                check_expected_value(line, VENDOR_ID, "Vendor ID")
+
+            elif "Product ID:" in line:
+                check_expected_value(line, PRODUCT_ID, "Product ID")
+
+            elif "Device Type:" in line:
+                check_expected_value(line, DEVICE_TYPE, "Device Type")
+
+            if "commissioner(s) discovered" in line:
+                break
+
+    # Wait for the processes to finish writing before attempting to read
+    time.sleep(PARSE_INTERVAL)
+
+    parse_linux_tv_casting_app_logs(LINUX_TV_CASTING_APP_LOGS)
```

Comment:
See comment above about succeeding fast. We probably will not need this timeout or separate file handling logic at all.

Reply:
Agreed, by succeeding fast, we do not need the timeout and/or separate parsing logic. Parsing will be done in real-time as we get the output from the tv-casting-app process. The change has been implemented.